docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,147 @@
+# Envoy Gateway Extensions Design
+
+As outlined in the [official goals][] for the Envoy Gateway project, one of the main goals is to "provide a common foundation for vendors to build value-added products 
+without having to re-engineer fundamental interactions". Development of the Envoy Gateway project has been focused on developing the core features for the project and 
+K8s Gateway API conformance. This system focuses on the “common foundation for vendors” component by introducing a way for vendors to extend Envoy Gateway.
+
+## Goals
+
+- Provide a foundation for extending the Envoy Gateway control plane
+- Allow users to attach [extensionRef HTTPRoute filters][] to resources introduced and handled by Envoy Gateway extensions.
+- Extension developers should **NOT** have to maintain a custom fork of Envoy Gateway
+- Extension developers should **NOT** require expert knowledge of all of Envoy Gateway's 
+internal components to meaningfully extend the control plane.
+- Provide a system for extending Envoy Gateway which allows extension projects to ship updates independent of Envoy Gateway's release schedule
+- Extensions can hook into both the Envoy configuration pipeline and the infra manager pipeline
+- Setup a foundation for the initial iteration of Extending Envoy Gateway
+
+## Non-Goals
+
+- Address every edge case for extending Envoy Gateway in the initial iteration
+- Extend [Gateway API Policy Attachments][]. At some point these will be addressed, but the initial implementation of the design omits these.
+
+## Overview
+
+TODO: I feel like this section could be better...
+
+Envoy Gateway provides a series of pre/post hooks before and after it's major components (translator, etc.) to allow extensions to modify the data going into or coming out of these components.
+An extension can be created external to Envoy Gateway as it's own deployment or loaded as a sidecar and will communicate over gRPC to receive data as well as context information and then reply 
+with modified data to be used by Envoy Gateway. As extensions fundamentally alter the logic and data that Envoy Gateway provides, Extension projects assume responsibility for any bugs and issues 
+they create as a direct result of their modification of Envoy Gateway.
+
+## Registering Extensions in Envoy Gateway
+
+Information about the extensions that Envoy Gateway needs to load is configured in the Envoy Gateway config. Each extension must specify:
+
+- The name to register the extension under
+- Which API groups the extension supports
```

Comment:
I think we will want to include the `Kinds` as well to make validation easier so the config could look like
```
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyGateway
extensions:
  - name: dummy-extension
    apiGroups:
      - apiGroup: example.myextension.io
        kinds:
          - MyFilter
    service:
      host: dummy-extension.example
      port: 443
      tls:
        type: secret
        secret:
            name: dummy-extension
            namespace: example
```

Reply:
I left that as a TODO comment in the POC implementation